Homework4/FtpServer/FtpServer.cs

```diff
@@ -0,0 +1,146 @@
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+
+namespace FtpServer;
+
+public class FtpServer
+{
+    private readonly FtpServerSettings _settings;
+    private readonly CancellationToken _token;
+
+    public FtpServer(FtpServerSettings settings, CancellationToken cancellationToken)
+    {
+        _settings = settings;
+        _token = cancellationToken;
+    }
+
+    private static async Task ListAsync(string path, NetworkStream stream)
+    {
+        if (!Directory.Exists(path))
+        {
+            await WriteStringToStreamAsync("-1", stream);
+            return;
+        }
+
+        var response = new StringBuilder();
+        var dirs = Directory.GetDirectories(path);
+        var files = Directory.GetFiles(path);
+
+        response.Append($"{dirs.Length + files.Length} ");
+        foreach (var dir in dirs) response.Append($"{dir} true ");
+
+        foreach (var file in files) response.Append($"{file} false ");
```

Comment:
```suggestion
        response.Append($"{dirs.Length + files.Length}");
        foreach (var dir in dirs) response.Append($" {dir} true");

        foreach (var file in files) response.Append($" {file} false");
```
Чуть поколдуем с пробелами и получим точное следование формату протокола (без лишнего пробела в конце). В задание сказано точно следовать протоколу - что если, чужой клиент не ожидает этот лишний пробел?
(-0.5)